module/localisation/RobotParticleLocalisation/src/RobotModel.hpp

```diff
@@ -95,7 +95,10 @@ namespace module::localisation {
 
             if (type == Goal::MeasurementType::CENTRE) {
                 const Eigen::Matrix<Scalar, 3, 1> rGCc(Hcf * actual_position);
-                return cartesianToSpherical(rGCc);
+
+                // Spherical Coordinates (1/distance, phi, theta)
+                auto measurement = cartesianToSpherical(rGCc);
```

Comment:
[auto for eigen types bad](https://eigen.tuxfamily.org/dox/TopicPitfalls.html)
```suggestion
                Eigen::Matrix<Scalar, 3, 1> measurement = cartesianToSpherical(rGCc);
```

Reply:
Adding an inverse function to make this mute as @TrentHouliston suggested

Reply:
I mean you still need to not use auto on it after you make the function :P

Reply:
Not if i don't make a temporary variable